### ISSUE TITLE
`bigram.py`'s training loop never prints the loss for the final iteration, unlike `gpt.py`

### ISSUE THREAD

**Issue by AzyzHm** [karpathy/ng-video-lecture#61]
**Description:**
`gpt.py`'s training loop explicitly makes sure the very last iteration's loss gets evaluated and printed:
```python
if iter % eval_interval == 0 or iter == max_iters - 1:
    losses = estimate_loss()
    print(f"step {iter}: train loss {losses['train']:.4f}, val loss {losses['val']:.4f}")
```
`bigram.py`'s otherwise-identical loop is missing the `or iter == max_iters - 1` part:
```python
if iter % eval_interval == 0:
    losses = estimate_loss()
    print(f"step {iter}: train loss {losses['train']:.4f}, val loss {losses['val']:.4f}")
```
With `bigram.py`'s defaults (`max_iters = 3000`, `eval_interval = 300`), the last training step is `iter = 2999`, and `2999 % 300 = 299 ≠ 0`, so this condition is never true on the final iteration — the last loss actually printed corresponds to `iter = 2700`, meaning the final ~300 training steps' effect on the loss is never shown before the script moves on to text generation. `gpt.py` (written later in the lecture series) fixed this; the fix apparently was never backported to the earlier, simpler `bigram.py`.
 
**Suggested fix:**
Match `gpt.py`'s condition:
```python
if iter % eval_interval == 0 or iter == max_iters - 1:
    losses = estimate_loss()
    print(f"step {iter}: train loss {losses['train']:.4f}, val loss {losses['val']:.4f}")
```
